tao_compiler/mlir/disc/transforms/shape_utils.cc

```diff
@@ -497,10 +556,51 @@ LogicalResult ShapeAnalysis::buildBlockDimValueMap(Block* block) {
       DimValue result(indexCast.getResult());
 
       if (failed(mapDimValueEqual(operand, result))) {
-        return WalkResult::interrupt();
+        return WalkResult::advance();
       }
       mayCreateOrMapConstInt(operand);
       mayCreateOrMapConstInt(result);
+    } else if (isa<mhlo::DynamicReshapeOp, mhlo::DynamicBroadcastInDimOp>(op)) {
```

Comment:
what are the rules to tell if an Op has to be put here,  for example, DynamicPad/DynamicIota? 
what will happen if they are not here?

Reply:
An op that generates shape value or utilizes shape value should be placed here. ShapeAnalysisV2 does not use such logic any more, which we can discuss offline.

Reply:
If they are not here, we may miss some shape information.